drivers/gpio/gpio_ite_it8xxx2.c

```diff
@@ -0,0 +1,474 @@
+				   [gpio_config->gpios_pin].reg_addr),
+				 GPIO_GPDRB_wui[gpio_config->gpios_pin].pin);
+		}
+
+	}
+	if (gpio_config->reg_addr == GPIO_GPDRF) {
+		SET_MASK(*(GPIO_GPDRF_wui[gpio_config->gpios_pin].reg_addr),
+			 GPIO_GPDRF_wui[gpio_config->gpios_pin].pin);
+		SET_MASK(*(GPIO_GPDRF_wui[gpio_config->gpios_pin].clear_addr),
+			 GPIO_GPDRF_wui[gpio_config->gpios_pin].pin);
+		if (both_tri_en == 1) {
+			SET_MASK(*(GPIO_GPDRF_wui
+				   [gpio_config->gpios_pin].reg_addr),
+				 GPIO_GPDRF_wui[gpio_config->gpios_pin].pin);
+		}
+	}
+
+	/* Enable IRQ */
+	irq_connect_dynamic(gpio_config->gpio_irq,
+				    0, gpio_ite_isr, dev, 0);
+	ite_intc_irq_enable(gpio_config->gpio_irq);
+
+	return ret;
+}
+
+static const struct gpio_driver_api gpio_ite_driver_api = {
+	.pin_configure = gpio_ite_configure,
+	.port_get_raw = gpio_ite_port_get_raw,
+	.port_set_masked_raw = gpio_ite_port_set_masked_raw,
+	.port_set_bits_raw = gpio_ite_port_set_bits_raw,
+	.port_clear_bits_raw = gpio_ite_port_clear_bits_raw,
+	.port_toggle_bits = gpio_ite_port_toggle_bits,
+	.pin_interrupt_configure = gpio_ite_pin_interrupt_configure,
+
+};
+
+
+static int gpio_ite_init(struct device *dev)
+{
+	const struct gpio_ite_cfg *gpio_config = DEV_GPIO_CFG(dev);
+	uint8_t regv;
+	unsigned int gpcr_offset = 0x10;
+	uint32_t gpcr_reg;
+	uint32_t gpcr_reg_addr;
+
+	if (gpio_config->reg_addr == GPIO_GPDRM) {
+		gpcr_offset = 0x40;
+	}
+	gpcr_reg = ((gpio_config->reg_addr & 0xff)  - 1) * 8 + gpcr_offset;
+	gpcr_reg_addr = gpcr_reg | (gpio_config->reg_addr & 0xffffff00);
+
+	if (gpio_config->flag & GPIO_OUTPUT) {
+		ite_write(gpcr_reg_addr + gpio_config->gpios_pin, 1, 0x40);
+		if (gpio_config->flag & GPIO_OUTPUT_INIT_HIGH) {
+
+			set_bit(gpio_config, gpio_config->gpios_pin, GPIO_HIGH);
+		} else if (gpio_config->flag & GPIO_OUTPUT_INIT_LOW) {
+			set_bit(gpio_config, gpio_config->gpios_pin, GPIO_LOW);
+		}
+	} else {
+
+		ite_write(gpcr_reg_addr + gpio_config->gpios_pin, 1, 0x80);
+		regv = ite_read(gpio_config->reg_addr, gpio_config->reg_size);
+	}
+	if (gpio_config->flag & GPIO_INT_ENABLE) {
+		gpio_pin_interrupt_configure(dev,
+					     gpio_config->gpios_pin,
+					     gpio_config->flag);
+	}
+
+	return 0;
+}
+
+#if DT_NODE_HAS_STATUS(DT_INST(0, gpio_leds), okay)
+static struct gpio_ite_data gpio_ite_data_led_0;
+static const struct gpio_ite_cfg gpio_ite_cfg_led_0 = {
+	.reg_addr = DT_REG_ADDR_BY_IDX(DT_PHANDLE_BY_IDX(
+					DT_NODELABEL(room_led), gpios, 0), 0),
+	.label = DT_LABEL(DT_PHANDLE_BY_IDX(
+				  DT_NODELABEL(room_led), gpios, 0)),
+	.gpios_pin = DT_PHA_BY_IDX(
+		DT_NODELABEL(room_led), gpios, 0, pin),
+	.flag = GPIO_OUTPUT_HIGH,
+};
+DEVICE_AND_API_INIT(gpio_it8xxx2_dev_led_0,
+		    DT_PROP(DT_NODELABEL(room_led), label),
+		    gpio_ite_init,
+		    &gpio_ite_data_led_0,
+		    &gpio_ite_cfg_led_0,
+		    POST_KERNEL,
+		    CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,
+		    &gpio_ite_driver_api);
+#endif          /* gpio_leds */
+
+
+#define GPIO_ITE_KEY_DEV_CFG_DATA(n) \
```

Comment:
This macro gets used exactly once, right below where it has been defined. The macro parameter 'n' is used in only one place in the macro. I think it would be easier and clearer to remove the macro and just use it inside the #if below.